### DISCUSSION TITLE
[Feature Request] unbind() function to stop listening on address/port

### DISCUSSION THREAD

**Discussion by TheCGDF** [actix/actix-web#3025] [Ideas]
Just like `iris` (a Golang web framework) does.

Perform hot reload automatically and safely, following these steps:

1. Listening for `IN_CLOSE_WRITE` signal for new version binary.
2. Replace old version binary with new one.
3. (`IN_CLOSE_WRITE` signal caught) Set executable permission to new version binary.
4. Stop listening immediately. (**Need this**)
5. Execute new version binary.
6. Wait and stop (`handle().stop(true)`) the old version binary

All steps can be done by myself except `4`

Is it possible to provide a function to stop listening immediately so that the address/port can be used by new version binary?

**Comment by izolyomi** [comment]
I've just encountered this as well. I tried to un/bind the port in a reliable app that reinitializes components and retries its whole workflow on any error encountered.

Unfortunatley, I've found no way to unbind the server port and this prevents me from properly reinitializing the application.

**Comment by robjtede** [comment]
this will be fixed by https://github.com/actix/actix-net/pull/966 in the upcoming actix-server release
